### PR TITLE
A brief notice for potential contributors, and brief documentation of…

### PR DESCRIPTION
**Pull request by pkofod**
… Particle Swarm.

Any additions or comments @anriseth ?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Notes for Optim Contributors
+
+Welcome to Optim.jl, and thanks for showing an interest in our package.
+
+## Before filing an issue
+
+- Reporting a potential bug? Please be sure to include a [Minimal Working Example](https://en.wikipedia.org/wiki/Minimal_Working_Example) so the other contributors can get right at solving your issue. Including the version of Optim.jl you have installed and the exact version of Julia you are using is encouraged as well.
+
+- Contributing code? We welcome all bug fixes, documentation, and new features. However, before you spend several weeks learning all the kinks and corners of Optim.jl to contribute a new algorithm, it is adviced to reach out via an issue or through [gitter](https://gitter.im/JuliaNLSolvers/Optim.jl) to gauge the interest.
```
Comment (anriseth) [inline CONTRIBUTING.md]: Add a header for contributing code?

File: docs/src/algo/particle_swarm.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+* `lower = []`, a vector of lower bounds, unbounded below if empty or `Inf`'s
+* `upper = []`, a vector of upper bounds, unbounded above if empty or `Inf`'s
+* `n_particles = 0`, number of particles in the swarm, defaults to least three
+
+## Description
+The Particle Swarm implementation in Optim.jl is the so-called Adaptive Particle
+Swarm algorithm in [1]. It attempts to improve global coverage and convergence by
+switching between four evolutionary states: exploration, exploitation, convergence,
+and jumping out. In the jumping out state it intentially tries to take the best
+particle and move it away from it's (potentially and probably) local optimum, to
```
Comment (anriseth) [inline docs/src/algo/particle_swarm.md]: it's -> its

File: docs/src/algo/particle_swarm.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Particle Swarm
+## Constructor
```
Comment (anriseth) [inline docs/src/algo/particle_swarm.md]: Add this as a docstring as well? (Maybe it already exists, I'm on my phone so I haven't checked)

Comment (pkofod) [reply docs/src/algo/particle_swarm.md]: Oh, it probably isn't, good idea!

**Comment by pkofod** [comment]
There are no "help wanted" issues yet, but let me add some tags ;)

**Comment by anriseth** [comment]
Ah, I guess the help wanted tags I've added were in some of the other repositories, then :p

**Pull request merged**